C7GameData/SaveFormat.cs

```diff
@@ -44,11 +76,30 @@ public static C7SaveFormat Load(string path)
             }
             return save;
         }
-        public static void Save(C7SaveFormat save, string path)
+
+        public static void Save(C7SaveFormat save, string path, SaveFileFormat format = SaveFileFormat.Plain)
         {
-            string json = JsonSerializer.Serialize(save, JsonOptions);
-            File.WriteAllText(path, json);
+            byte[] json = JsonSerializer.SerializeToUtf8Bytes(save, JsonOptions);
+            if (format == SaveFileFormat.Zip)
+            {
+                using (var zipStream = new MemoryStream())
+                {
+                    var archive = new ZipArchive(zipStream, ZipArchiveMode.Create);
+                    ZipArchiveEntry entry = archive.CreateEntry("save");
+                    using (Stream stream = entry.Open())
+                    {
+                        stream.Write(json, 0, json.Length);
+                    }
+                    archive.Dispose();
```

Comment:
It's _probably_ okay since you've tested it locally, but it looks suspicious that we're disposing of the ZipArchive before we write the data to disk.  Most likely it's probably in the MemoryStream already so it works... but if there isn't any benefit to doing this before writing to disk, I'd rather have code that doesn't look suspicious.

Reply:
It's unintuitive but we need to dispose of the archive in order for the contents of the stream to be valid to write to file: https://stackoverflow.com/questions/12347775/ziparchive-creates-invalid-zip-file#12350106

Reply:
Interesting.  Well, in that case, we'll have to keep it!  Maybe we should add a link to that answer in a comment above the line for the benefit of anyone investigating some future error who comes across the line; I agree it's unintuitive.